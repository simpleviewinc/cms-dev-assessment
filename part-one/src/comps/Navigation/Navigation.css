nav{
    display: flex;
    margin: 20px 0 0 10px;
}

.nav-button{
    margin: 0 6px;
    padding: 6px 12px;    
    text-decoration: none;
    background: #E4E4E4;
    color: #000;
    border-radius: 6px;
    border: 3px solid #E4E4E4;
}

.nav-button:hover, .nav-button:active{
    background: #AEAEAE;
    border: 3px solid #AEAEAE;
}

.nav-button:focus{
    background: #782BA7;
    color: #FFF;
    border: 3px solid #782BA7;
    box-shadow: inset 0px 0px 0px 2px #FFF;
    outline: none;

}