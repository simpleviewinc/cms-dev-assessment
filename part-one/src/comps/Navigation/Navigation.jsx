import React from "react"
import './Navigation.css'

export default class Navigation extends React.Component{

    handleAll = () => {
        this.props.handleAll()
    }
    
    handleListings = () => {
        this.props.handleListings()
    }
    
    handleEvents = () => {
        this.props.handleEvents()
    }

    handleOffers = () => {
        this.props.handleOffers()
    }

    render(){
        return(
            <nav>                
                <button 
                    onClick={this.handleAll}
                    className='nav-button'>
                        All
                </button>
                
                <button 
                    onClick={this.handleListings}
                    className='nav-button'>
                        Listings
                </button>
                
                <button 
                    onClick={this.handleEvents}
                    className='nav-button'>
                        Events
                </button>

                <button 
                    onClick={this.handleOffers}
                    className='nav-button'>
                        Offers                
                </button>                
            </nav> 
        )
    }
}