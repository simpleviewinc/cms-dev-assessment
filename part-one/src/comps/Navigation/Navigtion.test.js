import React from 'react';
import ReactDOM from 'react-dom';
import Navigation from './Navigation'
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Navigation />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected',() => {
  const tree = renderer
    .create(<Navigation />)
    .toJSON();
  expect(tree).toMatchSnapshot();  
  });