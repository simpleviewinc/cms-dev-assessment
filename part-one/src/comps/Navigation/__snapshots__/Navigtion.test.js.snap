// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<nav>
  <button
    className="nav-button"
    onClick={[Function]}
  >
    All
  </button>
  <button
    className="nav-button"
    onClick={[Function]}
  >
    Listings
  </button>
  <button
    className="nav-button"
    onClick={[Function]}
  >
    Events
  </button>
  <button
    className="nav-button"
    onClick={[Function]}
  >
    Offers
  </button>
</nav>
`;
